src/logic/router.js

```diff
@@ -0,0 +1,57 @@
+const get = require('lodash.get');
+const { wrap } = require('lambda-async');
+const Router = require('route-recognizer');
+const { wrap: wrapHandler } = require('./handler');
+const { ApiError } = require('../response');
+
+module.exports.Router = ({ module }) => {
+  const router = (() => {
+    const routerRec = new Router();
+    return {
+      register: (route, handler) => routerRec.add([{
+        path: route.split(/[\s/]/g).map((e) => e.replace(
+          /^{(.*?)(\+)?}$/,
+          (_, name, type) => `${type === '+' ? '*' : ':'}${name}`
+        )).join('/'),
+        handler
+      }]),
+      recognize: (method, path) => routerRec.recognize(`${method}${path}`)
+    };
+  })();
+
+  const handler = wrap(async (event, context) => {
+    const matchedRoutes = router.recognize(event.httpMethod, get(event, 'path', ''));
+    if (!matchedRoutes) {
+      const request = {
+        params: [],
+        options: {},
+        method: event.httpMethod,
+        uri: get(event, 'path', '')
+      };
+      const route = `${event.httpMethod} ${get(event, 'path', '')}`;
+      return wrapHandler(() => module.onUnhandled({
+        event,
+        context,
+        router
+      }).then((resp) => {
+        if (resp === null) {
+          throw ApiError('Method / Route not allowed', 403);
+        }
+        return resp;
+      }), {
+        request,
+        route,
+        router,
+        module
+      })(event, context);
+    }
+    return matchedRoutes[0].handler(Object.assign(event, {
```

Comment:
Destructuring cleans this one a bit
```suggestion
    return matchedRoutes[0].handler({
      ...event,
      pathParameters: matchedRoutes[0].params
    }, context);
```